server/controllers/invite.js

```diff
@@ -0,0 +1,61 @@
+const Invite = require("../models/Invite");
+const User = require("../models/User");
+const Team = require("../models/Team");
+const asyncHandler = require("express-async-handler");
+const sendEmailInvitation = require("../utils/sendEmailInvitation");
+
+exports.createInvitation = asyncHandler(async (req, res)=> {
+  const { senderId, recepientId, email, teamId } = req.body
+
+  const team = await Team.findById(teamId)
+  const sender = await User.findById(senderId)
+
+  if (!team || !sender) {
+    res.status(404);
+    throw new Error("Team and/or user could not be found");
+  }
+
+  if (email) {
+      sendEmailInvitation(email, sender.username, team.name);
+      const invite = await Invite.create({
+          sender,
+          email,
+          team
+      });
+      invite.save();
+      res.status(200).json(invite);
+  } else if (recepientId) {
+    const invite = await Invite.create({
```

Comment:
If recipientId is not given, isn't there a need to send a mail? _sendEmailInvitation_

Reply:
The email only sends if the req.body contains an email, if it contains a recepientId then I assumed that the invites would be fetched from the database automatically by the recepient.

Reply:
So, for a front-end implementation here would we be doing the search for a user's email and passing it down, I mean ultimately this would be either their ID or email. 

I think it would have been better to just pass in the email check for a user with that email in the database and then either way we are sending an email invitation.

Reply:
Essentially, a user wouldn't search by an ID but rather a name or email